diff -Naur libwebsockets-4.2.0.orig/CMakeLists.txt libwebsockets-4.2.0/CMakeLists.txt
--- libwebsockets-4.2.0.orig/CMakeLists.txt	2021-04-15 16:44:02.000000000 +0000
+++ libwebsockets-4.2.0/CMakeLists.txt	2021-05-08 18:59:08.350508559 +0000
@@ -366,6 +366,10 @@
 		set(LWS_BUILD_HASH ${GIT_USER}@${GIT_HOST}-${GIT_HASH})
 	endif()
 
+	if ("${LWS_BUILD_HASH}" STREQUAL "")
+		set(LWS_BUILD_HASH "unknown")
+	endif()
+
 	message("Git commit hash: ${LWS_BUILD_HASH}")
 endif()
 
