diff -Naurp a/psplash-fb.c b/psplash-fb.c
--- a/psplash-fb.c	2020-08-12 10:44:48.000000000 -0700
+++ b/psplash-fb.c	2021-06-29 21:09:52.885448257 -0700
@@ -553,7 +553,7 @@ psplash_fb_text_size (int
   mbtowc (0, 0, 0);
   for (; (k = mbtowc (&wc, c, n)) > 0; c += k, n -= k)
     {
-      if (*c == '\n')
+      if (*c == '\v')
 	{
 	  if (w > mw)
 	    mw = w;
@@ -592,7 +592,7 @@ psplash_fb_draw_text (PSplashFB
     {
       u_int32_t *glyph = NULL;
 
-      if (*c == '\n')
+      if (*c == '\v')
 	{
 	  dy += h;
 	  dx  = 0;
